geode-core/src/main/java/org/apache/geode/security/ResourcePermission.java

```diff
@@ -145,10 +145,10 @@ private void init(String resource, String operation, String target, String key)
     }
 
     setParts(Arrays.asList(
-        Collections.singleton(this.resource),
-        Collections.singleton(this.operation),
-        Collections.singleton(this.target),
-        Collections.singleton(this.key)));
+        CollectionUtils.asSet(this.resource.split(SUBPART_DIVIDER_TOKEN)),
+        CollectionUtils.asSet(this.operation.split(SUBPART_DIVIDER_TOKEN)),
```

Comment:
Since resource and operation are derived from enums I don't think it is possible to have those sub-divided.